### ISSUE TITLE
cli/demo: stray files left over in /tmp after Ctrl+C

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#46988]
Because the `demo` code relies on the Go runtime's SIGHUP handler, Ctrl+C during the shell's initialization, or in-between SQL statements (i.e. not at the interactive prompt) leaves files over, for example:

```
$ ls /tmp
cockroach-auxiliary580965477/
cockroach-auxiliary644807582/
demo964434555/
```

This is because the transient cluster's stopper is not stopped in that case, nor is the temp dir cleanup function called.

The solution for this would be to register a signal handler and let it dance gracefully with the one used by libedit at the prompt.

Jira issue: CRDB-6326

**Comment by knz** [comment]
cc @rohany  (not urgent)

**Comment by knz** [comment]
Fixed by #76433
